### PR TITLE
Release notes for pyMOR 0.5

### LINKED ISSUE #505: Write release notes for 0.5 release

### PR DESCRIPTION
**Pull request by sdrave**
Fixes #505.

### CODE REVIEW

**Comment by sdrave** [comment]
Dear @pymor/pymor-devs, please take all a few minutes to read through (and improve) our release notes. @pmli, maybe you could make the sys-mor section a little bit more prosaic like the other sections? 

If all goes well, we could have a release tomorrow morning!

File: docs/source/release_notes.rst [hunk 1]
```diff
@@ -4,11 +4,322 @@
...
+- The somewhat dubious concept of a `subtype` has been superseded by the concept
+  of |VectorSpaces| which act as factories for |VectorArrays|. In particular,
+  instead of a `subtype`, |VectorSpaces| can now hold meaningful attributes
+  (e.g. the dimension) which are required to construct |VectorArrays| contained
+  in the space. The
+  :attr:`~pymor.vectorarrays.interfaces.VectorSpaceInterface.id` attribute
+  allows to differentiate between technically identical but mathematically
+  different spaces `[#323] <https://github.com/pymor/pymor/pull/323>`_.
+
+- |VectorArrays| can now be indexed to select a subset to vectors to operate on.
```
Comment (ftalbrecht) [inline docs/source/release_notes.rst]: subset _to_ vectors -> subset _of_ vectors

File: docs/source/release_notes.rst [hunk 2]
```diff
@@ -4,11 +4,322 @@
...
 *************
 
+pyMOR 0.5 (January ??, 2019)
+----------------------------
+
+After more than two years of development, we are proud to announce the release
+of pyMOR 0.5! Highlights of this release are support for Python 3, bindings for
+the NGSolve finite element library, new linear algebra algorithms, various
+|VectorArray| usability improvements, as well as a redesign of pyMOR's
+projection algorithm based on |RuleTables|.
```
Comment (ftalbrecht) [inline docs/source/release_notes.rst]: algorithm -> algorithms

File: docs/source/release_notes.rst [hunk 3]
```diff
@@ -4,22 +4,340 @@
...
+Several structure-preserving MOR methods for second-order systems have been
+implemented. Balancing-based MOR methods are implemented in
+:mod:`pymor.reductors.sobt`, bitangential Hermite interpolation in
+:class:`~pymor.reductors.interpolation.SO_BHIReductor` and Second-Order Reduced
+IRKA (SOR-IRKA) in :class:`~pymor.reductors.sor_irka.SOR_IRKAReductor`.
+
+For more general transfer functions, MOR methods which return |LTISystems| are
+also available. Bitangential Hermite interpolation is implemented in
+:class:`~pymor.reductors.interpolation.TFInterpReductor` and Transfer Function
+IRKA (TF-IRKA) in :class:`~pymor.reductors.h2.TF_IRKAReductor`.
```
Comment (ftalbrecht) [inline docs/source/release_notes.rst]: Very nice. Maybe mention the `heat` demo?

**Pull request merged**